pandas/core/arrays/period.py

```diff
@@ -228,6 +303,183 @@ def start_time(self):
     def end_time(self):
         return self.to_timestamp(how='end')
 
+    def __repr__(self):
+        return '<{}>\n{}\nLength: {}, dtype: {}'.format(
```

Comment:
dont' we have a default repr that should work here?

Reply:
Not yet, but we should indeed have it. See also https://github.com/pandas-dev/pandas/issues/22846 
But let's put this on the list of follow-up issues?